### PR TITLE
Grover class crosslinks note to tutorials and textbook 

### PR DESCRIPTION
**Pull request by 1ucian0**
A note in the Grover class to crosslink to textbook and tutorial.

### CODE REVIEW

File: qiskit/algorithms/amplitude_amplifiers/grover.py [hunk 1]
```diff
@@ -29,6 +29,12 @@
 class Grover(AmplitudeAmplifier):
     r"""Grover's Search algorithm.
 
+        .. note::
+
+            If you want to learn more about Grover's Search algorithm, checkout the
+            `Qiskit Textbook <https://qiskit.org/textbook/ch-algorithms/grover.html>`_.
+            or the Qiskit Tutorials
```
Comment (jakelishman) [inline qiskit/algorithms/amplitude_amplifiers/grover.py:36]:
```suggestion
            If you want to learn more about Grover's Search algorithm, check out the
            `Qiskit Textbook <https://qiskit.org/textbook/ch-algorithms/grover.html>`_.
            or the Qiskit Tutorials.
```

**Review by jakelishman — COMMENTED** [review]
Looks fine to me in theory.

lol at Matthew and Kevin getting tagged by `qiskit-bot` - I think it's because "amplitude_amplifiers" matches the regex `"*rs"` haha.

File: qiskit/algorithms/amplitude_amplifiers/grover.py [hunk 2]
```diff
@@ -143,7 +149,7 @@ def __init__(
 
         if growth_rate is not None:
             # yield iterations ** 1, iterations ** 2, etc. and casts to int
-            self._iterations = map(lambda x: int(growth_rate**x), itertools.count(1))
+            self._iterations = map(lambda x: int(growth_rate ** x), itertools.count(1))
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply qiskit/algorithms/amplitude_amplifiers/grover.py:152]: oh.. shoot. I was in the wrong env. thanks

**Comment by 1ucian0** [comment]
> Looks fine to me in theory.

The idea is start moving towards a cleaner API reference, where the details lives somewhere else.

File: qiskit/algorithms/amplitude_amplifiers/grover.py [hunk 3]
```diff
@@ -29,6 +29,13 @@
 class Grover(AmplitudeAmplifier):
     r"""Grover's Search algorithm.
 
+        .. note::
+
+            If you want to learn more about Grover's Search algorithm, check out the
+            `Qiskit Textbook <https://qiskit.org/textbook/ch-algorithms/grover.html>`_.
+            or the `Qiskit Tutorials
+            <https://qiskit.org/documentation/tutorials/algorithms/07_grover_examples.html>`_.
```
Comment (woodsp-ibm) [inline qiskit/algorithms/amplitude_amplifiers/grover.py:37]: The textbook link, to me, is more if you are interested in the theory and how to build out such an algorithm yourself, whereas the latter is more instructional around using this specific implementation here. I don't know if its worth qualifying the text in the this regard. Linking them with `or` above might imply equivalence at some level, but they are kinda different.  One can be interested in both aspects of course,

**Comment by mtreinish** [comment]
> lol at Matthew and Kevin getting tagged by qiskit-bot - I think it's because "amplitude_amplifiers" matches the regex "*rs" haha.

Heh, I guess I'll have to update the regex to be a bit more exact than what's there now.

File: qiskit/algorithms/amplitude_amplifiers/grover.py (hunk 3 above)
Comment (Cryoris) [reply qiskit/algorithms/amplitude_amplifiers/grover.py:37]: Like so?
```suggestion
            If you want to learn more about the theory behind Grover's Search algorithm, check 
            out the `Qiskit Textbook <https://qiskit.org/textbook/ch-algorithms/grover.html>`_.
            or the `Qiskit Tutorials
            <https://qiskit.org/documentation/tutorials/algorithms/07_grover_examples.html>`_
            for more concrete how-to examples.
```

Comment (1ucian0) [reply qiskit/algorithms/amplitude_amplifiers/grover.py:37]: sounds good

**Review by jakelishman — APPROVED** [review]

**Pull request merged**